hasura/graphql-engine
# Propagate the trace context to event triggers

This enables the collection of traces which span a mutation and the event trigger(s) it invokes.

### Changelog


### Affected components



### Related Issues




### Solution and Design

- The trace context is written into a transaction-local variable `hasura.trace_context`
- This is picked up by the `insert_event_log` function and written into the `event_log` table
- If a event contains a trace context, that trace is extended. Otherwise, a new one is created.

### Steps to test and verify

- Create an event trigger
- Perform an insert mutation
- Observe the trace context stored on the event: `SELECT * FROM hdb_catalog.event_log WHERE payload -> 'trace_context' IS NOT NULL`

### Limitations, known bugs & workarounds



### Server checklist


#### Catalog upgrade

Does this PR change Hasura Catalog version?

#### Metadata


Does this PR add a new Metadata feature?


#### GraphQL
#### Breaking changes
